app/graphql/mutations/update_school_link.rb

```diff
@@ -8,23 +8,48 @@ def validate(_object, _context, value)
       end
     end
 
+    class TitleConditionallyRequired < GraphQL::Schema::Validator
+      def validate(_object, _context, value)
+        title = value[:title]
+        kind = value[:kind]
+        if title.blank? && kind != SchoolLink::KIND_SOCIAL
+          return(I18n.t("mutations.update_school_link.blank_title_error"))
+        end
+      end
+    end
+
+    class ValidTitleLength < GraphQL::Schema::Validator
+      def validate(_object, _context, value)
+        title = value[:title]
+        return unless title
+
+        if title.length < 1 || title.length > 24
+          return(
+            I18n.t("mutations.update_school_link.invalid_title_length_error")
+          )
+        end
+      end
+    end
```

Comment:
Can we move this into a shared validation as its the same for create and edit

Reply:
I had refactored and moved the validation into a shared validation